### PR TITLE
Events COMPSERV-45 COMPSERV-46 COMPSERV-48 COMPSERV-79 COMPSERV-80

### PR DESCRIPTION
**Pull request by bri25yu**
- Added functionality to allow creation of recurring events on calendar
- Added separation by types for All RSVPs page
- Added markdown capabilities to event descriptions
- Added clickable link functionality to event locations
- Added functionality for officers to confirm on the events detail page
- Added access level permissions functionality to events

compserv hknweb events details page changes
compserv hknweb events edit page changes
compserv hknweb events new page changes
compserv hknweb events all rsvps page changes

### CODE REVIEW

File: hknweb/events/templates/events/event_form_base.html [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+        });
+        $(function () {
+            $("#id_end_time").datetimepicker({
+                format: 'm/d/Y g:i A',
+                formatTime: 'g:i A',
+                validateOnBlur: false,
+            });
+        });
+    </script>
+</div>
```
Comment (conhuang) [inline hknweb/events/templates/events/event_form_base.html:81]: This is amazing, thank you for making the code more concise

File: hknweb/events/models.py [hunk 2]
```diff
@@ -22,8 +24,16 @@ class Event(models.Model):
     end_time    = models.DateTimeField()
     location    = models.CharField(max_length=255)
     event_type  = models.ForeignKey(EventType, models.CASCADE)
-    description = models.TextField()
+    description = MarkdownxField()
     rsvp_limit  = models.PositiveIntegerField(null=True, blank=True)
+    access_level = models.IntegerField(
```
Comment (Rishabhk2357) [inline hknweb/events/models.py:29]: Is there a level for officers/should there be?

File: hknweb/events/templates/events/event_form_base.html (hunk 1 above)
Comment (bri25yu) [reply hknweb/events/templates/events/event_form_base.html:81]: yeet!

File: hknweb/events/models.py (hunk 2 above)
Comment (bri25yu) [reply hknweb/events/models.py:29]: I originally thought there should be a distinction between officers and members, but then I was wondering how often (if at all) we would ever need that distinction (like are there any officer events that members shouldn't see?). What do you think?

Comment (bri25yu) [reply hknweb/events/models.py:29]: Closing for now! (we'll probably end up doing access control in a less terrible way later on too lmao)

Comment (Rishabhk2357) [reply hknweb/events/models.py:29]: Yeah this makes a lot of sense (sorry this slipped by).

**Pull request merged**